homeassistant/components/olarm/coordinator.py

```diff
@@ -0,0 +1,333 @@
+"""The coordinator for the olarm integration to handle API and MQTT connections."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+import ssl
+
+from aiohttp import ClientResponseError
+from olarmflowclient import OlarmFlowClient, OlarmFlowClientApiError
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryNotReady
+from homeassistant.helpers import config_entry_oauth2_flow
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class OlarmFlowClientCoordinator:
```

Comment:
What is this object? It's not a conventional coordinator as we have them as a common pattern. Can you elaborate?

Do you get a full state for the full system or just parts?